### DISCUSSION TITLE
Learn front-end technologies or not?

### DISCUSSION THREAD

**Discussion by LifeProgramming** [Cyfrin/foundry-full-course-cu#2380] [Q&A] [answered]
Hi Guys! I have recently finished the Solidity foundry smart contract development and want to start working as a smart contract and dApp developer. I am familiar with some front-end technologies like JavaScript, HTML and CSS. My question: Is it necessary to learn front-end frameworks completely like React to become a dApp developer or I can do it without any front-end tools( Like backend developer in web development who can do his role without working in required front-end role)?

**Comment by EngrPips** [comment, marked as answer]
You said `want to start working as a smart contract and dApp developer` so I assume that is some sort of `full-stack` stuff so you should be pretty good in front-end and back-end (smart contract development) in my opinion.

**Reply by LifeProgramming** [reply]
Does it mean that I should work with front-end technologies if I want to develop dApps( beside of the smart contract) and I can work with just Solidity if I want to develop just the platform's smart contract?

**Reply by EngrPips** [reply]
You can use your HTML, CSS, and Javascript knowledge to develop a dapp if that will be easy for you when developing a complex dapp.
